test/issues/general/test_9399.test

```diff
@@ -0,0 +1,31 @@
+# name: test/issues/general/test_9399.test
+# description: Issue 9399: Incorrect query output from group by query (regression in 0.9.0)
+# group: [general]
+
+# happened with > 8 threads
+statement ok
+SET threads=10
+
+# seed so it's deterministic
+statement ok
+SELECT SETSEED(0.8765309);
+
+# create table that's has just one row group, and just a few duplicates
+statement ok
+CREATE TABLE test AS
+SELECT CASE WHEN random() < 0.002 THEN range ELSE 100000 - range END c0,
+    random() c1,
+    random() c2
+FROM range(100000)
+
+mode output_hash
```

Comment:
Can we remove the `mode output_hash`?

Reply:
Of course! I think this is ready to go now